core/web/user_controller.go

```diff
@@ -37,6 +37,42 @@ func (c *UserController) UpdatePassword(ctx *gin.Context) {
 	}
 }
 
+// NewAPIToken generates a new API token for a user overwriting any pre-existing one set.
+func (c *UserController) NewAPIToken(ctx *gin.Context) {
```

Comment:
Idea: This stretches the focus of `UserController` and a `user` resource, resource meaning it's a single entity with CRUD operations. Perhaps we could just go ahead and make a new resource/controller with `APIController`/`TokensController` which has its own CRUD, or in this case, just CreateToken and DeleteToken.

Reply:
The tokens are added into the user model though, so they are operating on user model data, wouldn't that mean they user this controller?  How else should the relationship between the User and API tokens be represented on the web layer?